tests/test_parallelreduce.py

```diff
@@ -40,5 +44,53 @@ def test_add_squares(self):
         self.assertEqual(expected_result, result)
 
 
+@pk.workload
+class SquareSumFloat:
+    def __init__(self, n):
+        self.N: int = n
+        self.total: pk.double = 0
+
+    @pk.main
+    def run(self):
+        self.total = pk.parallel_reduce(self.N, self.squaresum)
+
+    @pk.workunit
+    def squaresum(self, i: float, acc: pk.Acc[pk.double]):
+        acc += i * i
+
+
+@pk.workload
+class SquareSumInt:
+    def __init__(self, n):
+        self.N: int = n
+        self.total: pk.int64 = 0
+
+    @pk.main
+    def run(self):
+        self.total = pk.parallel_reduce(self.N, self.squaresum)
+
+    @pk.workunit
+    def squaresum(self, i: int, acc: pk.Acc[pk.int64]):
+        acc += i * i
+
+
+@pytest.mark.parametrize("series_max", [10, 5000, 90000])
+@pytest.mark.parametrize("dtype", [np.float64, np.int64])
+def test_squaresum_types(series_max, dtype):
+    # related to gh-6
+
+    # Enforce pykokkos reduction return view type
+    # matches accumulator type, indirectly via
+    # consistency with NumPy for lack
+    # of overflow, etc.
+    expected = np.sum(np.arange(series_max, dtype=dtype) ** 2)
+    if dtype == np.float64:
+        ss_instance = SquareSumFloat(series_max)
+    elif dtype == np.int64:
+        ss_instance = SquareSumInt(series_max)
+    pk.execute(pk.ExecutionSpace.OpenMP, ss_instance)
+    actual = ss_instance.total
+    assert_allclose(actual, expected)
```

Comment:
This test exposes some overflow/typing issues that still remain as I note above, more than it really focuses on the reduction view type. Actually, I'm not sure we have a great handle for checking the reduction view type in the Python layer, since we only get the reduced `self.total` scalar back?

Nonetheless, I suspect we may agree that this test should be possible to make pass eventually.